src/AutoMapper.Extensions.ExpressionMapping/XpressionMapperVisitor.cs

```diff
@@ -306,7 +306,9 @@ Expression DoVisitUnary(Expression updated)
         protected override Expression VisitConstant(ConstantExpression node)
         {
             if (this.TypeMappings.TryGetValue(node.Type, out Type newType)
-                && ConfigurationProvider.ResolveTypeMap(node.Type, newType) != null)
+                && ConfigurationProvider.ResolveTypeMap(newType, node.Type) != null)
```

Comment:
This can't be correct.  The code now says:
```c#
	if (this.TypeMappings.TryGetValue(node.Type, out Type newType)
		&& ConfigurationProvider.ResolveTypeMap(newType, node.Type) != null)
	{//if map newType -> oldType exists then map oldType to newType
		return base.VisitConstant(Expression.Constant(Mapper.MapObject(node.Value, node.Type, newType), newType));
	}
```

The configuration requirement is different for mapping objects within the expression.  For mapping expressions you do need the reverse.

Your mappings worked before because the object value was null (the constant object may not be null).

Your options are one of the following:

- Add the mapping.
- Separate the null check from non-null mapping.

```c#
            if (this.TypeMappings.TryGetValue(node.Type, out Type newType))
            {
                if (node.Value == null)
                    return base.VisitConstant(Expression.Constant(null, newType));

                if (ConfigurationProvider.ResolveTypeMap(node.Type, newType) != null)
                    return base.VisitConstant(Expression.Constant(Mapper.MapObject(node.Value, node.Type, newType), newType));
            }
```

Reply:
Sorry about the formatting changes, those weren't intended, both Visual Studio and SourceTree didn't bother to show them in the diff.

I've implemented the second option, as adding the mapping isn't truly viable for our use case. We map filter expressions from our DTOs to our database models and most times will have a few different kinds of mapping.

In this scenario ViewModelDTO inherits PutModelDTO and we have the following mappings:
- Entity to ViewModelDTO
- Entity to PutModelDTO
- PutModelDTO to Entity 

Most times, filter expressions are done on the ViewModelDTO and then mapped on, implementing a mapping from ViewModelDTO to the Entity is impractical for both the amount of work (legacy database reasons) and from a code maintainability standpoint.
All our filter expressions work generally with some extra null checks on subtypes and eventually value type/string equality or comparison, which is why I didn't even consider reference type mapping and its paths in this scenario. Sorry about that and good catch!

Thanks for your help.